[Carmen Ang] iP

cmang12 opened the pull request:
### Toto 🐈‍⬛
> With great power comes great responsibility ([source](https://en.wikipedia.org/wiki/With_great_power_comes_great_responsibility#:~:text=%22With%20great%20power%20comes%20great,to%20the%20young%20Peter%20Parker.))

Toto is your friendly neighbourhood cat who happens to be able to manage your tasks! Toto is,

- text-based
- super intuitive
- speedy
- and the bestest friend! 

What are you waiting for? Go ahead and, 

1. download the jar file 
2. double-click it.
3. add your tasks.
4. and allow Toto to manage them for you. 

**YOUR** life will now be: 

- [X] easy
- [X] hassle-free
- [X] convenient

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```
public static void main(String[] args) {
        new Duke("src/main/data/duke.txt", "src/main/data").run();
}
```

Commit:
Add Gradle support

Commit:
added greeting and echo function

Commit:
added list function

Commit:
added list function

Commit:
added Task class and can mark/unmark tasks

Commit:
added todo, event, deadline classes and functionality

Commit:
edited expected and input txt files for automated ui testing

Commit:
added exception handling

Commit:
added delete function

Commit:
added save functionality

Commit:
added LocalDateTime as format for time for Deadline and Event class

Commit:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

Commit:
fixed errors after the merge with branch-Level-7 and branch-Level-8

Commit:
added more OOP

Commit:
added packages

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'

Commit:
Add J-Unit tests

There are no unit testing previously.

Adding unit testing for each Task class (TodoTest.java, EventTest.java, Deadline.java) and TaskList.java is so each method for each class can be isolated and tested to make sure that it is working properly.

Commit:
Added Java Doc

JavaDoc is to generate standard documentation in HTML format and gives concise descriptions of functions.

Java documentation headers are written for all classes' methods.

Commit:
Update coding standard

Some methods were not named appropriately. Imports did not have the correct conventions.

Methods in Ui class are now named more appropriately according to the Java coding standard so that their actions are clearer by name.

Imports at the beginning of file are sorted in alphabetical order and spaced out. Doing so makes it easier to browse through the list of imports especially when there are many.

Commit:
Add find function

User can now input a find command and the program lists out tasks in the task. If task cannot be found, then it will throw an error message otherwise, it lists tasks with that keyword.

Adding this function allows user to better navigate and find the task list.

Commit:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Ui.java

Commit:
Added javadoc

Commit:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/TaskList.java

Commit:
Added javadoc

Added javadoc for new methods in Parser class.

Commit:
Added checkstyle.

Previous code had issues with style.

Adding checkstyle allows to see errors in style of code.

Doing so can makes the code look cleaner and more standard.

Commit:
Add CheckStyle

Commit:
Add GUI

tituswe commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,123 @@
+                    LocalDateTime to = parser.getEventTo(askTo);
+                    ui.showAddTask(tasks.addEvent(description, from, to));
+                    continue;
+                }
```
You could consider using enums and/or polymorphism and inheritance to process which commands to execute. This way you wouldn't have to keep adding new cases for new commands and your `main` class would be a lot cleaner!

tituswe commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,192 @@
+ */
+public class Parser {
+
+    protected String answer;
```
Could the field name `answer` be slightly ambiguous? You could consider a name such as 'argument' or 'args'. 👍

tituswe commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,124 @@
+ * Handles the storage of tasks into the user's computer.
+ */
+public class Storage {
+    protected String filePath;
+    protected String dirPath;
```
I like the field naming used here - I used the same thing!

tituswe commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,64 @@
+     * Returns status of task in 1 or 0.
+     * @return Status of task in 1 or 0.
+     */
+    public String getStatusIconBinary() {
+        return (isDone ? "1" : "0");
+    }
```
I like the distinction you made from the `getStatusIcon()` method.

tituswe commented on src/test/java/duke/task/DeadlineTest.java:
```diff
@@ -0,0 +1,26 @@
+package duke.task;
+import org.junit.jupiter.api.Test;
```
Should you leave a newline after the package? Just pointing it out incase you missed it.

tituswe commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,136 @@
+     * Prints out note after a task is deleted by the user.
+     * @param task Task that is deleted.
+     */
+    public void deleted(Task task) {
```
Should you change this method name to `sayDeleted` to conform with the other methods in your code?

tituswe commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,136 @@
+    /**
+     * Prints out exit note after the user inputs "bye".
+     */
+    public void goodbye() {
```
Should you change this method name to `sayGoodbye` to conform with the other methods in your code?

tituswe reviewed (commented):
Mostly great code quality, especially for naming conventions. All the best!

Commit:
Add GUI to Duke

Previous implementation of Duke was only text.

By adding GUI, user experience is now enhanced. Now user is able to see icons and have a more interactive UI.

JavonTeo commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,64 @@
+/**
+ * Task class.
+ */
+public class Task {
```
Could the Task class be extracted out, perhaps as an abstract class instead?

JavonTeo commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,157 @@
+     * @return Deleted Task.
+     */
+    public Task delete(Integer index) {
+        Task t = arrayList.get(index - 1);
```
Would the variable name t be ambiguous? Perhaps a more intuitive name such as "task" could be used. I noticed the same issue in several other places too.

JavonTeo reviewed (commented):
Overall, the code was easy to read. I noticed a few minor coding standard violations. Other than that, seems it LGTM!

Commit:
Add Assertions

- Previous implementation does not use assertions.
- Assertions are added to handle the user commands when they input a task.
- Assertions are used so that runtime can detect any errors or wrong assumptions.

Commit:
Improve Code Quality

There are changes to the code quality mostly towards shortening the code and making the happy path prominent.

This is so that the code is more readable and the path where everything goes well is clear.

cmang12 committed:
Merge pull request #2 from cmang12/branch-A-Assertions

Add Assertions

Commit:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/Duke.java

cmang12 committed:
Merge pull request #3 from cmang12/branch-A-CodeQuality

Improve Code Quality

Commit:
no message

Commit:
Add Archive Function

User can now archive all tasks and deleted tasks will go to the archive list. This is so that the user can obtain have a clean slate in their main task list.

cmang12 committed:
Merge pull request #4 from cmang12/branch-C-Archive

Add Archive Function

Commit:
Add Better GUI and personality

- GUI is modified to be better looking.
- To add more personality, phrases and pictures are also modified.
- Better handling for empty lists.

Commit:
Add exit command

The user now enters save to save the current task list and enters exit to exit the program.

Commit:
Update JavaDoc

Commit:
Add UI screenshot

cmang12 committed:
Update README.md

cmang12 committed:
Update README.md

cmang12 committed:
Update README.md

Commit:
Add Javadoc

Previous version had the wrong format for Javadoc in some methods.

It is now changed so that the descriptons for each method is clearer and concise.

Commit:
Change mainclass
